From 9abe1a6ca0895bfdee9b7c4c7114c10c65631160 Mon Sep 17 00:00:00 2001
From: Cyrill Gorcunov <gorcunov@openvz.org>
Date: Wed, 18 Mar 2009 18:48:33 +0300
Subject: [PATCH] bc: fix permissions on /proc/bc

The reading of /proc/bc/* is permitted for those only who
has CAP_DAC_OVERRIDE and CAP_DAC_READ_SEARCH attributes
set. We should not point files as "group" or "other"
readable/executable since they are not.

Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 kernel/bc/proc.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/bc/proc.c b/kernel/bc/proc.c
index c78d744..b3d7178 100644
--- a/kernel/bc/proc.c
+++ b/kernel/bc/proc.c
@@ -423,7 +423,7 @@ static int bc_lookset(struct inode *ino, void *data)
 	ino->i_ino = bc_make_ino(ub);
 	ino->i_fop = &bc_entry_fops;
 	ino->i_op = &bc_entry_iops;
-	ino->i_mode = S_IFDIR | S_IRUSR | S_IXUGO;
+	ino->i_mode = S_IFDIR | S_IRUSR | S_IXUSR;
 	/* subbeancounters are not included, but who cares? */
 	ino->i_nlink = num_entries + 2;
 	ino->i_gid = 0;
@@ -661,7 +661,7 @@ static int __init ub_init_proc(void)
 	struct proc_dir_entry *entry;
 
 	bc_proc_root = create_proc_entry("bc",
-			S_IFDIR | S_IRUGO | S_IXUGO, NULL);
+			S_IFDIR | S_IRUSR | S_IXUSR, NULL);
 	if (bc_proc_root == NULL)
 		panic("Can't create /proc/bc entry");
 
@@ -675,7 +675,7 @@ static int __init ub_init_proc(void)
 	bc_register_proc_root_entry(&bc_all_resources_entry);
 
 	entry = proc_create("user_beancounters",
-			S_IRUGO, &glob_proc_root, &ub_file_operations);
+			S_IRUSR, &glob_proc_root, &ub_file_operations);
 	return 0;
 }
 
-- 
1.6.2

